client/sites-dashboard/components/sites-table-row.tsx

```diff
@@ -79,6 +88,16 @@ const VisitDashboardItem = ( site: Site ) => {
 
 export default function SitesTableRow( { site }: SiteTableRowProps ) {
 	const { __ } = useI18n();
+
+	const isComingSoon =
+		site.is_coming_soon || ( site.is_private && site.launch_status === 'unlaunched' );
+	let icon = globe;
```

Comment:
I generally think `let` are code smells (there's even an ESLint rule to prefer `const`) and honestly, this case deserves much more than an `if`. It makes a lot of sense to extract this whole block to a function that has a `switch` statement inside, for example. Taking decisions inside a React component, that accounts mostly for the UI, doesn't feel right.